compiler/src/iree/compiler/Codegen/Common/test/reconcile_translation_info_linearize.mlir

```diff
@@ -0,0 +1,176 @@
+// RUN: iree-opt --split-input-file --pass-pipeline="builtin.module(hal.executable(hal.executable.variant(iree-codegen-reconcile-translation-info{distribute-along=z}, canonicalize)))" --allow-unregistered-dialect --mlir-print-local-scope %s | FileCheck %s --check-prefix=DISTRIBUTEZ
+
+#pipeline_layout = #hal.pipeline.layout<constants = 6, bindings = [
+    #hal.pipeline.binding<storage_buffer>]>
+hal.executable private @scf_forall_2D_dynamic_tile_size {
+  hal.executable.variant public @scf_forall_2D_dynamic_tile_size target(#hal.executable.target<"", "", {}>) {
+    hal.executable.export public @scf_forall_2D_dynamic_tile_size layout(#pipeline_layout) count(%arg0: !hal.device, %arg1: index, %arg2 : index, %arg3 : index, %arg4 : index, %arg5 : index, %arg6 : index) -> (index, index, index) {
+      %x, %y, %z = iree_tensor_ext.dispatch.workgroup_count_from_slice %arg1, %arg2, %arg3, %arg4, %arg5, %arg6
+      hal.return %x, %y, %z : index, index, index
+    }
+    builtin.module {
+      func.func @scf_forall_2D_dynamic_tile_size() {
+        %cst0 = hal.interface.constant.load layout(#pipeline_layout) ordinal(0) : index
+        %cst1 = hal.interface.constant.load layout(#pipeline_layout) ordinal(1) : index
+        %cst2 = hal.interface.constant.load layout(#pipeline_layout) ordinal(2) : index
+        %cst3 = hal.interface.constant.load layout(#pipeline_layout) ordinal(3) : index
+        %cst4 = hal.interface.constant.load layout(#pipeline_layout) ordinal(4) : index
+        %cst5 = hal.interface.constant.load layout(#pipeline_layout) ordinal(5) : index
+        %0 = iree_tensor_ext.dispatch.workload.ordinal %cst0, 0 : index
+        %1 = iree_tensor_ext.dispatch.workload.ordinal %cst1, 1 : index
+        %2 = iree_tensor_ext.dispatch.workload.ordinal %cst2, 2 : index
+        %3 = iree_tensor_ext.dispatch.workload.ordinal %cst3, 3 : index
+        %4 = iree_tensor_ext.dispatch.workload.ordinal %cst4, 4 : index
+        %5 = iree_tensor_ext.dispatch.workload.ordinal %cst5, 5 : index
+        scf.forall (%arg0, %arg1) = (%0, %1) to (%2, %3) step(%4, %5) {
+          "use"(%arg0, %arg1) : (index, index) -> ()
+          scf.forall.in_parallel {}
+        } {mapping = [#iree_codegen.workgroup_mapping<y>, #iree_codegen.workgroup_mapping<x>]}
+        return
+      }
+    }
+  }
+}
+//       DISTRIBUTEX:   hal.executable.export
+//  DISTRIBUTEX-SAME:       %[[ARG0:[a-zA-Z0-9]+]]: !hal.device
+//  DISTRIBUTEX-SAME:       %[[ARG1:[a-zA-Z0-9]+]]: index
+//  DISTRIBUTEX-SAME:       %[[ARG2:[a-zA-Z0-9]+]]: index
+//  DISTRIBUTEX-SAME:       %[[ARG3:[a-zA-Z0-9]+]]: index
+//  DISTRIBUTEX-SAME:       %[[ARG4:[a-zA-Z0-9]+]]: index
+//  DISTRIBUTEX-SAME:       %[[ARG5:[a-zA-Z0-9]+]]: index
+//  DISTRIBUTEX-SAME:       %[[ARG6:[a-zA-Z0-9]+]]: index
```

Comment:
Here is the trick to remove dup checks: you can add `CHECK-ALL` or `DISTRIBUTE-ALL` to the `--check-prefix`s. Then you can write checks like:

```mlir
// CHECK-ALL:       hal.executable.export
// CHECK-ALL-SAME:    %[[ARG0:[a-zA-Z0-9]+]]: !hal.device
// CHECK-ALL-SAME:    %[[ARG1:[a-zA-Z0-9]+]]: !hal.device

...

// The chunk of X checks
// DISTRIBUTEX: ...

// The chunk of Y checks.
// DISTRIBUTEY: ...

// The chunk of Z checks:
// DISTRIBUTEZ: ...
```

Below is an example in IREE:

https://github.com/iree-org/iree/blob/d1aaeae079bf689eb895a1b6599f0988892fb162/compiler/src/iree/compiler/Codegen/Common/test/decompose_pack_unpack_ops.mlir#L29-L41